### ISSUE TITLE
[Feature request] Add option to sign in to Google account (as workaround for "Please sign in" blocking playback)

### ISSUE THREAD

**Issue by retrocaster** [iv-org/invidious#4977]
I would like to suggest adding an optional ability to sign in to a Google account, purely as a way to get around the "Please sign in" errors which are now occurring for every video I try to play despite hosting my own private instance on a residential ip address.

I realize this goes against the privacy goal of Invidious, but that's not why I use Invidious. I use Invidious because it allows me to play ad free Youtube on my Apple TV via Yattee. I don't care if Google is tracking what I watch and I suspect there are other users that feel the same.

**Is your feature request related to a problem? Please describe.**
Countless bugs have been filed on the issue of Youtube blocking playback. For example, https://github.com/iv-org/invidious/issues/4975

**Describe the solution you'd like**
Add an optional way to log into a Google account using username/password

Thanks for any consideration!

**Comment by SamantazFox** [comment]
We removed the "login with Google" feature in #3893 because their 2FA process and account protection is way too complicated for us to follow.

However, you can connect to the official website, then copy your sessions cookies and put them in your instance's `config.yml`:
https://github.com/iv-org/invidious/blob/53e8a5d62d4d7d66e8819f221a1c7b168102995c/config/config.example.yml#L157-L163

**Comment by unixfox** [comment]
> However, you can connect to the official website, then copy your sessions cookies and put them in your instance's `config.yml`:

There are more things to do in order to have this working. People explained what in the private matrix room.

> We removed the "login with Google" feature in #3893 because their 2FA process and account protection is way too complicated for us to follow.

Youtube oauth2 makes this quite easy, in fact.

**Comment by retrocaster** [comment]
> There are more things to do in order to have this working. People explained what in the private matrix room.
> 
> > We removed the "login with Google" feature in #3893 because their 2FA process and account protection is way too complicated for us to follow.
> 
> Youtube oauth2 makes this quite easy, in fact.

Can you post steps here, or do I need to find them on the private matrix room? This seems like it would be useful info to have in the official documentation.

**Comment by bbsixzz** [comment]
Yes please, I'm rotating po_token like crazy every day ever since this started.

**Comment by unixfox** [comment]
This will be considered as an option in a future change. For now there are unofficial ways getting shared on matrix to get this working, but we have no plans to share them here since this is not the correct way to logging into a google account. Doing it the way being shared on matrix can result in the logging not working after a few days and requiring to logging every week for example.

Anyway we want to make it clear, this is not the ideal solution. People on yt-dlp for example have said that they were getting their google account banned pretty quickly by using it on yt-dlp.

**Comment by ddscentral** [comment]
Well, worst of all, not even po_token works now. Even with a residential IP...
Having a way to login (say with a disposable account) would certainly help.

**Comment by unixfox** [comment]
That's totally offtopic @veganomy. Please use matrix or IRC for that: https://matrix.to/#/#invidious:matrix.org or https://web.libera.chat/?channel=#invidious

This issue is about logging to Google, please stay on topic.

**Comment by unixfox** [comment]
Invidious companion support was merged into the master branch, it allows to logging into a Google account. BUT it's not usable because:

- This bug: https://github.com/iv-org/invidious-companion/issues/37
- Google detects the usage of logged account and shadow-ban the account. After a few hours your account can't load youtube videos anymore.
- The YouTube client type that we are using is now encrypting video streams using DRM and companion does not support DRM.

We are unsure if we will be able to provide this logging as a Google account support due to the DRM usage. Maybe we will try to not use oauth, but instead have some kind of dialog that launch a real browser in order to logging into a google account. I don't know yet.

But all future request about logging into a Google account will now need to happen on the dedicated repository of Invidious companion: https://github.com/iv-org/invidious-companion/issues
